android/java/org/chromium/chrome/browser/playlist/kotlin/util/HLSParsingUtil.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package org.chromium.chrome.browser.playlist.kotlin.util
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.net.Uri
+import android.util.Log
+import androidx.media3.exoplayer.hls.playlist.HlsMediaPlaylist
+import androidx.media3.exoplayer.hls.playlist.HlsMultivariantPlaylist
+import androidx.media3.exoplayer.hls.playlist.HlsPlaylistParser
+import java.io.File
+import java.io.FileInputStream
+import java.util.LinkedList
+import java.util.Queue
+
+@SuppressLint("UnsafeOptInUsageError")
+@Suppress("unused")
+object HLSParsingUtil {
+    private val TAG: String = "Playlist/" + this::class.java.simpleName
+
+    @JvmStatic
+    fun getContentManifestUrl(
+        context: Context,
+        baseUrl: String,
+        localManifestFilePath: String
+    ): String {
+        var contentManifestUrl = ""
+        val newBaseUrl = Uri.parse(baseUrl).buildUpon().clearQuery().build().toString()
+        try {
+            val hlsParser =
+                context.contentResolver?.openInputStream(Uri.parse(localManifestFilePath))
```

Comment:
it would be good to create a follow up for adding asserts for all IO operations as @bridiver mentioned:
`ThreadUtils.assertOnBackgroundThread();`

Reply:
https://github.com/brave/brave-browser/issues/42777